File: 383. 赎金信/Solution.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] record = new int[26];
        for(char i: magazine.toCharArray()){
            record[i - 'a'] += 1;
        }
        for(char i: ransomNote.toCharArray()){
            record[i - 'a'] -= 1;
            if(record[i - 'a'] < 0) return false;
        }
        return true;
    }
}

File: 383. 赎金信/Solution.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dict1 = {}
        for i in magazine:
            if i not in dict1.keys():
                dict1[i] = 1
            else:
                dict1[i] += 1
        for i in ransomNote:
            if i not in dict1.keys() or dict1[i] == 0:
                return False
            else:
                dict1[i] -= 1
        return True
